### PR TITLE
Add support for fabric monitor daemon (swss part).

### PR DESCRIPTION
**Pull request by jfeng-arista**
Add support for fabric monitor daemon ( swss related changes ) 

Added fabricmgr that reacts the change in config_db and writes to appl_db, as orchagent can only reacts to the appl_db changes in current infrastructure.   

So, when users configure the fabric link monitoring polling period or threshold to a none default values, the cli changes the value in config_db. This fabricmgr will react to the change in config_db and copy the value to appl_db. The fabric orchagent can react to the appl_db change and adjust the algorithm accordingly. 

need to merge https://github.com/sonic-net/sonic-sairedis/pull/1301 and https://github.com/sonic-net/sonic-buildimage/pull/16791 first for test.

**What I did**

**Why I did it**

**How I verified it**

**Details if related**

### CODE REVIEW

**Comment by jfeng-arista** [comment]
this depends on https://github.com/sonic-net/sonic-swss-common/pull/808/ . we need submodule update .

**Comment by prsunny** [comment]
@arlakshm , please review

**Comment by prsunny** [comment]
Please add unit-tests

**Comment by jfeng-arista** [comment]
https://github.com/sonic-net/sonic-sairedis/pull/1301 is the preparation for adding tests for this PR

**Comment by jfeng-arista** [comment]
For running the test, we need to merge 
https://github.com/sonic-net/sonic-sairedis/pull/1301  and
https://github.com/sonic-net/sonic-buildimage/pull/16791

**Comment by jfeng-arista** [comment]
the vs test failures needs
https://github.com/sonic-net/sonic-sairedis/pull/1301 
and 
https://github.com/sonic-net/sonic-buildimage/pull/16791
merge first

File: cfgmgr/fabricmgr.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    isolateStatus = fvValue(i);
+                    writeConfigToAppDb(key, "isolateStatus", isolateStatus);
+                }
+                else
+                {
+                    field_values.emplace_back(i);
+                }
+            }
+
+            for (auto &entry : field_values)
```
Comment (arlakshm) [inline cfgmgr/fabricmgr.cpp:90]: why do we need this?

File: tests/test_fabric_port.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        dvss = vst.dvss
+        for name in dvss.keys():
+            dvs = dvss[name]
+            # Get the config info
+            config_db = dvs.get_config_db()
+            metatbl = config_db.get_entry("DEVICE_METADATA", "localhost")
+
+            cfg_switch_type = metatbl.get("switch_type")
+            if cfg_switch_type == "fabric":
```
Comment (arlakshm) [inline tests/test_fabric_port.py:22]: do we need to test if `switch_type == VOQ` ?

File: cfgmgr/fabricmgr.cpp (hunk 1 above)
Comment (jfeng-arista) [reply cfgmgr/fabricmgr.cpp:90]: > why do we need this?

This is needed for adding the fields in to the appl_db. other mgr , like portmgr , also do the same thing. thank you

File: tests/test_fabric_port.py (hunk 2 above)
Comment (jfeng-arista) [reply tests/test_fabric_port.py:22]: > do we need to test if `switch_type == VOQ` ?

In the current vs test environment, the fabric related setup ( ports, counters and etc. ) are all under the cfg_switch_type fabric. So I am having this if check here. 

If you really want to start testing the VOQ, that will need some more infra changes, if so, I can do in a separate PR later to cover them there if you are ok.  thank you

**Review by arlakshm — DISMISSED** [review]

**Comment by jfeng-arista** [comment]
https://github.com/sonic-net/sonic-sairedis/pull/1301 merged in on  Sun, Oct 8
now we need to wait https://github.com/sonic-net/sonic-buildimage/pull/16791   to run the stest.

**Comment by arlakshm** [comment]
Hi @jfeng-arista https://github.com/sonic-net/sonic-buildimage/pull/16791 is merged now. Can you help update the branch.

**Comment by saksarav-nokia** [comment]
@jfeng-arista , I am adding the fabric monitoring device files for nokia platform and i am trying to understand the lanes field in the fabric_port_config.ini file.  Does this need to match to the SAI_PORT_ATTR_HW_LANE_LIST returned by SAI for the fabric port?.
Could please confirm?
# name              lanes     isolateStatus
Fabric0             0         False
Fabric1             1         False
Fabric2             2         False
Fabric3             3         False
Fabric4             4         False
Fabric5             5         False

Thanks,
Sakthi

**Comment by jfeng-arista** [comment]
test_drop_counters.py:762: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_drop_counters.TestDropCounters object at 0x7f3a31c208e0>
stats = ['SAI_PORT_STAT_IN_CONFIGURED_DROP_REASONS_0_DROPPED_PKTS', 'SAI_PORT_STAT_IN_CONFIGURED_DROP_REASONS_1_DROPPED_PKTS']
counter_list_type = 'PORT_DEBUG_COUNTER_ID_LIST'

    def checkFlexState(self, stats, counter_list_type):
        flex_counter_table = swsscommon.Table(self.flex_db, FLEX_COUNTER_TABLE)
    
        for oid in flex_counter_table.getKeys():
            attributes = self.genericGetAndAssert(flex_counter_table, oid)
            assert len(attributes) == 1
            field, tracked_stats = attributes[0]
            assert field == counter_list_type
            for stat in stats:
>               assert stat in tracked_stats
E               AssertionError: assert 'SAI_PORT_STAT_IN_CONFIGURED_DROP_REASONS_1_DROPPED_PKTS' in 'SAI_PORT_STAT_OUT_CONFIGURED_DROP_REASONS_1_DROPPED_PKTS,SAI_PORT_STAT_IN_CONFIGURED_DROP_REASONS_0_DROPPED_PKTS'

test_drop_counters.py:109: AssertionError


this is the new test failure, not seems related to this change, will try trigger the test pipeline again

**Comment by jfeng-arista** [comment]
Now all tests passed

**Review by arlakshm — APPROVED** [review]

**Comment by prsunny** [comment]
@jfeng-arista , @arlakshm , could you please confirm all dependent PRs are merge and this is ready for merge?

**Comment by jfeng-arista** [comment]
> @jfeng-arista , @arlakshm , could you please confirm all dependent PRs are merge and this is ready for merge?

Yes. thank you

**Pull request merged**